### DISCUSSION TITLE
The last step of installation error

### DISCUSSION THREAD

**Discussion by rogerluo0101** [eclipse-syson/syson#83] [Q&A] [answered]
Hi guys,
 I build and install the syson(in Windows) following the guide from : https://www.youtube.com/watch?v=IgFEfofNCbM 
After a long time of build, I successfully build the project:
![image](https://github.com/eclipse-syson/syson/assets/5074071/7dc31b22-b9cb-4b17-ae4f-885e427ae366)
But the last step of launching docker comes to error and can not continue(I confirmed I had logged in the docker):
![image](https://github.com/eclipse-syson/syson/assets/5074071/bb962437-bdfa-41d7-a674-129e1d3b7ab8)
Can anyone help me to figure out what's wrong with my steps:

My Steps:
Precondition: install git/nodejs/maven3.8.1/docker and set the environment variable correctly
1. git clone https://github.com/eclipse-syson/syson.git
2.npm login --scope=@NAMESPACE  --auth-type=legacy --registry=https://npm.pkg.github.com/ 
 personal access token (classic):[Managing your personal access tokens - GitHub Docs](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) 
3.npm config set script-shell "C:\Program Files\Git\bin\bash.exe"
4.npm ci
5.npx turbo run build
6."C:\Program Files\Git\bin\bash.exe"
7.mkdir -p backend/application/syson-frontend/src/main/resources/static
cp -R frontend/syson/dist/* backend/application/syson-frontend/src/main/resources/static
8.exit
9.set USERNAME=<github_username>
set PASSWORD=<token>
mvn -U -B -e clean verify --settings settings.xml
10.cd backend/application/syson-application
docker-compose up
11.localhost:8080

My docker version is :Docker version 25.0.2, build 29cf629

**Comment by AxelRICHARD** [comment, marked as answer]
Hello,
This video is not from the SysOn team so I can't help you on this.

But, If you want to test the current implementation, an easy way to do it is :

1 - Download the jar of the application from https://github.com/eclipse-syson/syson/packages/2020337. You should see a syson-application-2024.1.0.jar (the version number will increase over time) file on the "assets" section of the page. This is a single "fat jar" which contains the complete SysON application with all its dependencies.

2 - Install docker and create a postgres db. You should follow the same instructions than Sirius Web (https://github.com/eclipse-sirius/sirius-web?tab=readme-ov-file#quick-start) but adapted for SysON :

```
docker run -p 5433:5432 --name syson-postgres \
                            -e POSTGRES_USER=dbuser \
                            -e POSTGRES_PASSWORD=dbpwd \
                            -e POSTGRES_DB=syson-db \
                            -d postgres:12
```

3 - Start the application

```
java -jar syson-application-2024.1.0.jar \
         --spring.datasource.url=jdbc:postgresql://localhost:5433/syson-db \
         --spring.datasource.username=dbuser \
         --spring.datasource.password=dbpwd \
         --spring.liquibase.change-log=classpath:db/changelog/sirius-web.db.changelog.xml (note that sirius-web.db.changelog.xml is used here (not syson.db.changelog.xml) , because SysON relies on the Sirius Web Persistence module)
```

4 - Point your browser at http://localhost:8080/ and enjoy!

If you encounter some difficulties, please read the https://github.com/eclipse-sirius/sirius-web?tab=readme-ov-file#quick-start. Some notes and warnings may help you to solve them.

An official documentation with installation & build instructions should be available soon.

Regards,

**Reply by rogerluo0101** [reply]
HI Richard
          I deployed the  syson-application-2024.1.0.jar and use localhost:8080 and I can visit the mainpage. But if I replaced the "localhost" to my local IP(192.168.0.XX:8080), it does not work. I guess it related to the network configuration of the tomcat server, How can I change the "localhost" to "server IP" so that everyone can visit the page from LAN? Thank you very much. Looking forward to your reply.

**Comment by rogerluo0101** [comment]
Thank you very much, it works.
![image](https://github.com/eclipse-syson/syson/assets/5074071/f00ab6dc-c0b4-4199-83df-788c9d0efb76)

**Comment by rogerluo0101** [comment]
HI Richard
I deployed the syson-application-2024.1.0.jar and use localhost:8080 and I can visit the mainpage. But if I replaced the "localhost" to my local IP(192.168.0.XX:8080), it does not work. I guess it related to the network configuration of the tomcat server, How can I change the "localhost" to "server IP" so that everyone can visit the page from LAN? Thank you very much. Looking forward to your reply.

**Reply by AxelRICHARD** [reply]
Hello,
Do you have any stack trace/error message to share from the backend console/frontend console?
Regards,
Axel

**Reply by stephanelacrampe** [reply]
FYI I tried the same, I have the localhot:8080 working nicely, but if I enter anything else (10.0.0.250 is my local ip address), I get a blank page. Nothing in the backend console. If I do a "View Source" from the blank page in the browser, I can see this message:     <noscript>You need to enable JavaScript to run this app.</noscript>
(which is obviously enabled since it is working with the localhost url)

**Reply by AxelRICHARD** [reply]
Hello, after analyzed the problem on my side and discussed it with the team, it is due to `Crypto.randomUUID` which is used in Sirius Web and SysON. This method only works in a secure context (please read https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts for more details). Localhost is considered a secure context, but not 192.168.X.X.
So, It means if you want your SysON application to be accessible from remote, you should consider enable https on your server (which probably means have a certificate, and have some specific configuration on your server).
Regards,

**Reply by rogerluo0101** [reply]
Hi Richard,
         I follow this guide to create a self-cert and verfify “https://192.168.XX.XX" correctly: https://akshitb.medium.com/how-to-run-https-on-localhost-a-step-by-step-guide-c61fde893771
         But I think this command should give me an option to load the cert since the tomcat server is packaged in the jar.
“java -jar syson-application-2024.1.0.jar  --spring.datasource.url=jdbc:postgresql://localhost:5433/syson-db  --spring.datasource.username=dbuser  --spring.datasource.password=dbpwd --spring.liquibase.change-log=classpath:db/changelog/sirius-web.db.changelog.xml”
        I don't know if I understand correct? Thank you for your reply.

**Reply by AxelRICHARD** [reply]
Hello,
I don't know if the link you provided provides the appropriate solution for the problem.
May be try to search in spring documentation if there are option related to https/ssl/tls configuration.
I'm sorry, but I can't dig further on the subject as this is not a priority for the team at the moment.
Regards,
Axel
